torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1186,7 +1244,12 @@ Tensor* TensorExprKernel::computeValue(const torch::jit::Value* v) {
 
     case aten::abs: {
       return computeOneOperand(
-          "aten_abs", v, [](const ExprHandle& a) { return fabs(a); });
+          "aten_abs",
+          v,
+          [](const ExprHandle& a) {
+            return fabs(promoteHalfToFloatAndIntegerToDefaultType(a));
```

Comment:
This doesn't seem like the best solution for abs -
in eager we're we run abs(input) 
Here in NNC we're calling int(fabs(float(input)) which is more work & likely slower. What is motivating this change ?

Reply:
I am working on a fix to add integers to `abs`. Still debugging it (as `SimpleIREvaluator` right now assumes intrinsic are always floating point types, `abs` will be the first one that truly supports integer types), but will submit a PR soon.

Reply:
Regarding allowing `half` for `IntrinsicsOps`, we had a discussion on it in earlier reviews. We all agree w/ each other. But I propose to fix it in a separate issue and have opened #48864 to track it (I can work on it after my current set of issues land).

And regarding supporting integers for `kAbs`, it is tracked in #48860 (which I have started working on it last Friday, but got delayed because of debugging the unexpected CI failure in the PR :<).

Reply:
Cool! yea i think we're all agreed on best solution ! I guess I was just wondering whether it would be better to separately fix the integral evaluation and the half-rewriting than adding the extra casts/potentially changing numerics in the short term. It would be nice to fix the IREvaluator but when supporting it leaks into effecting codegen that seems not great. I'm fine doing whatever though.

 The link I posted adds an integral intrinsic - https://github.com/pytorch/pytorch/pull/48973/files#diff-b800236a0a3b78031df12c064de7b33e4a72f5d076a7353f3d8f5a60160b0e15R901. You might try something similar for `abs`. You'll soon find that no CI failure is all that unexpected 😛